uring/jvm/src/main/scala/fs2/io/uring/net/UringSocketGroup.scala

```diff
@@ -131,84 +134,82 @@ private final class UringSocketGroup[F[_]: LiftIO](implicit F: Async[F], dns: Dn
 
         sockets = Stream
           .resource(createBufferAux(isIpv6))
-          .flatMap { buf =>
+          .flatMap { buf => // Buffer that will contain the remote address
             Stream
-              .resource(createBuffer(4))
-              .flatMap { buf2 =>
-                Stream.resource {
-
-                  buf2.writeInt(buf.capacity())
-
-                  val accept =
-                    Resource.eval(F.delay(println("[SERVER] accepting connection..."))) *>
-                      ring
-                        .bracket(
-                          IORING_OP_ACCEPT,
-                          0,
-                          NativeAccess.SOCK_NONBLOCK,
-                          linuxSocket.fd(),
-                          buf.memoryAddress(),
-                          0,
-                          buf2.memoryAddress()
-                        )(closeSocket(ring, _))
-                        .mapK {
-                          new cats.~>[IO, IO] {
-                            def apply[A](ioa: IO[A]) = ioa.debug()
+              .resource(createBuffer(4)) // TODO: Replace 4 with INT_SIZE ?
+              .flatMap {
+                bufLength => // ACCEPT_OP needs a pointer to a buffer containing the size of the first buffer
+                  Stream.resource {
+
+                    bufLength.writeInt(buf.capacity())
+
+                    // We accept a connection, we write the remote address on the buf and we get the clientFd
+                    val accept =
+                      Resource.eval(F.delay(println("[SERVER] accepting connection..."))) *>
+                        ring
+                          .bracket(
+                            op = IORING_OP_ACCEPT,
+                            rwFlags = SOCK_NONBLOCK,
```

Comment:
I've been confused about this before, but I think we actually shouldn't use the `SOCK_NONBLOCK` flag when opening a socket.

- https://github.com/armanbilge/fs2-io_uring/pull/58

Reply:
Oh Interesting... Should we use the alternative `SOCK_CLOEXEC`?

Reply:
tbh I don't really know what that flag does 😅 

currently on Native we aren't passing any flags.
https://github.com/armanbilge/fs2-io_uring/blob/cc293e400d4f9572566474405a4a07898ef32ae3/uring/src/main/scala/fs2/io/uring/net/UringSocketGroup.scala#L117-L121

Reply:
Apparently, "it is use to ensure that if any child process is created it won't inherit this socket." I guess we don't need any flag for the moment